test/githubWrapperTests.js

```diff
@@ -60,11 +79,114 @@ describe('githubwrapper.js Tests', function() {
 
     it('should return a promise from getUser', function() {
         var username = 'testUser';
-        var result = gitHubWrapper.getUser(username);
+        var result = gitHubWrapper.getUser(username, {});
 
         expect(result).to.have.property('then');
     });
 
+    it('authenticate should use token type auth when username is `<token>`', (done) => {
+        const password = 'thisisanapitoken';
+        sandbox.stub(inquirer, 'prompt').resolves({username: '<token>', password});
+        gitHubWrapper.authenticate().then(() => {
+            expect(mockGithub.authenticate.calledOnce).to.be.true;
+            expect(mockGithub.authenticate.args[0][0]).to.eql({type: 'token', token: password});
+            done();
+        })
+        .catch(e => {
+            done(e);
+        });
+    });
+
+    it('authenticate should use basic type auth when username is anything except `<token>`', (done) => {
+        const password = 'this is some password';
+        const username = 'not-token';
+        sandbox.stub(inquirer, 'prompt').resolves({username, password});
+        gitHubWrapper.authenticate().then(() => {
+
+            expect(mockGithub.authenticate.calledOnce).to.be.true;
+            expect(mockGithub.authenticate.args[0][0]).to.eql({type: 'basic', username, password});
+            done();
+        })
+        .catch(e => {
+            done(e);
+        });
+    });
+
+    describe('2FA checks', () => {
+        let errorSpy;
+        const otpError = {
+            code: 401,
+            headers: { ['x-github-otp']: 'required; app'}
+        };
+        const otherError = new Error('not a 2FA issue');
+
+        const otpReject = (spy) => {
+            spy.reset();
+            spy.rejects(otpError);
+        };
+        const otherReject = (spy) => {
+            spy.reset();
+            spy.rejects(otherError);
+        };
+
+        beforeEach(() => {
+            errorSpy = sandbox.stub(console, 'error');
+        });
+
+        [
+            ['getUser', mock => mock.users.getForUser],
+            ['getMembers', mock => mock.orgs.getMembers],
+            ['getTeamMembers', mock => mock.orgs.getTeamMembers],
+            ['getTeams', mock => mock.orgs.getTeams],
+            ['getCollaborators', mock => mock.repos.getCollaborators],
+            ['getOrganizations', mock => mock.users.getOrgs],
+        ].forEach(([methodName, getApiSpy]) => {
```

Comment:
If `getDefaultBranch` gets the `checkFor2faError` treatment then `getRepoInfo` should be tested as well.

Reply:
getRepoInfo doesn't need it. The only thing it does is a shell command to list remotes and call getDefaultBranch. the only ones I tested and added the check for were exported methods that interacted with the github api library. I'll add an export for getDefaultBranch and add the check.